ext/GeometryOpsLibGEOSExt/simplify.jl

```diff
@@ -0,0 +1,8 @@
+# Address potential ambiguities
+GO._simplify(::GI.PointTrait, ::GO.TopologyPreserve, geom; kw...) = geom
+GO._simplify(::GI.MultiPointTrait, ::GO.TopologyPreserve, geom; kw...) = geom
+
+function GO._simplify(::GI.AbstractGeometryTrait, alg::GO.TopologyPreserve, geom)
```

Comment:
Why don't we have to pass in `:GEOS` here like we do in `simple_overrides.jl`? I get that we don't have a native way of doing that yet, but I worry that people who want to use the `GO.TopologyPreserve` algorithm won't even realize that is it calling `LibGEOS`. Perhapes this needs to be emphasized in comments here and (stronger) in comments in `simplify.jl`.

Reply:
Should this have errors like buffer about LibGEOS needing to be loaded?

Reply:
GEOS has a regular `simplify` which is the equivalent of DouglasPeucker here, this is a different algorithm.  I can definitely add this to the docstring for the algorithm struct, though.

Reply:
Yeah this is a little odd to add, especially when the other simplify methods all have keywords that `TopologyPreserve` doesn't have.

It would def need a "Load LibGEOS" warning.

Reply:
I think its best not to add this and let us to add our own TopologyPreseve later that doesn't mean GEOS topology preserve.

Reply:
Oh make it so you need to use `GEOS()` now as well as `TopologyPreserve`